server-common/src/main/java/org/apache/kafka/server/metrics/KafkaYammerMetrics.java

```diff
@@ -78,8 +102,10 @@ public void validateReconfiguration(Map<String, ?> configs) throws ConfigExcepti
 
     @Override
     public void reconfigure(Map<String, ?> configs) {
-        Predicate<String> mBeanPredicate = JmxReporter.compilePredicate(configs);
-        jmxReporter.updatePredicate(metricName -> mBeanPredicate.test(metricName.getMBeanName()));
+        if (jmxReporter != null) {
+            Predicate<String> mBeanPredicate = JmxReporter.compilePredicate(configs);
+            jmxReporter.updatePredicate(metricName -> mBeanPredicate.test(metricName.getMBeanName()));
+        }
```

Comment:
Do we need to cover that case where the JMX reporter is initially configured and then dynamically removed and therefore needs to be `shutdown()`? And the other way around too?